### PR TITLE
Added initial clear chat flags tests

### PR DESCRIPTION
**Pull request by TheHollidayInn**
These are the clear flags tests for #6509. We might want to use my new generate chat function when that PR gets merged, but I feel pretty comfortable using these tests as well.

### CODE REVIEW

File: website/src/controllers/api-v3/chat.js [hunk 1]
```diff
@@ -256,4 +257,51 @@ api.flagChat = {
...
+    }
+
+    message.flagCount = 0;
+
+    await Group.update(
+      {_id: group._id, 'chat.id': message.id},
+      {$set: {'chat.$.flagCount': message.flagCount}}
+    );
+
+    res.respond(204, message.flagCount);
```
Comment (paglias) [inline website/src/controllers/api-v3/chat.js]: 204 means no content. but for compatibility with mobile apps we prefer to always return an empty object so I'd do `res.respond(200, {})`

File: website/src/controllers/api-v3/chat.js [hunk 2]
```diff
@@ -256,4 +257,51 @@ api.flagChat = {
...
+    let validationErrors = req.validationErrors();
+    if (validationErrors) throw validationErrors;
+
+    let group = await Group.getGroup({user, groupId});
+    if (!group) throw new NotFound(res.t('groupNotFound'));
+
+    let message = _.find(group.chat, {id: chatId});
+    if (!message) throw new NotFound(res.t('messageGroupChatNotFound'));
+
+    if (!user.contributor.admin) {
```
Comment (paglias) [inline website/src/controllers/api-v3/chat.js]: I'd move it right after parameter validation so we don't fetch the group if user is not an admin

File: website/src/controllers/api-v3/chat.js (hunk 1 above)
Comment (TheHollidayInn) [inline website/src/controllers/api-v3/chat.js]: Should I still return the flag count?

**Comment by TheHollidayInn** [comment]
Updated!

File: website/src/controllers/api-v3/chat.js [hunk 3]
```diff
@@ -256,4 +257,51 @@ api.flagChat = {
...
+/**
+ * @api {post} /groups/:groupId/chat/:chatId/clear-flags Clear a group chat message's flags
+ * @apiVersion 3.0.0
+ * @apiName ClearFlags
+ * @apiGroup Chat
+ *
+ * @apiParam {groupId} groupId The group _id
+ * @apiParam {chatId} chatId The chat message _id
+ *
+ * @apiSuccess {Array} chat An array of chat messages
```
Comment (crookedneighbor) [inline website/src/controllers/api-v3/chat.js]: We're not returning an array of chat messages in this route.

**Comment by TheHollidayInn** [comment]
Fixed the note.

**Pull request merged**